src/emu/diimage.cpp

```diff
@@ -1309,10 +1309,9 @@ void device_image_interface::update_names()
 			count++;
 	}
 
-	m_canonical_instance_name = string_format("%s%d", inst_name, index + 1);
 	if (count > 1)
 	{
-		m_instance_name = m_canonical_instance_name;
```

Comment:
It is my understanding that having briefinstance name conflicts is an illegal condition.  If this isn't caught by the validity checks, it should be.  Moreover, this issue (while important) is orthogonal to the options overhaul.